python/pyarrow/tests/test_compute.py

```diff
@@ -1360,6 +1361,27 @@ def test_filter_null_type():
     assert len(table.filter(mask).column(0)) == 5
 
 
+@pytest.mark.parametrize("ty", ["inclusive"])
+def test_between_array_array_scalar(ty):
+    BetweenOptions = partial(pc.BetweenOptions)
+
+    val = pa.array([1, 1, 4, 3, 2, 5])
+    arr1 = pa.array([1, 2, 3, 2, None, 2])
+    scalar2 = pa.scalar(4)
+
+    inclusive_and_expected = {
+        "both": [True, False, True, True, None, False],
+        "left": [True, False, False, True, None, False],
+        "right": [False, False, True, True, None, False],
+        "neither": [False, False, False, True, None, False],
+    }
+
+    for inclusive, expected in inclusive_and_expected.items():
+        options = BetweenOptions(inclusive=inclusive)
+        result = pc.between(val, arr1, scalar2, options=options)
```

Comment:
You might just as well pass the option directly:
```suggestion
        result = pc.between(val, arr1, scalar2, inclusive=inclusive)
```